### ISSUE TITLE
Vaguely Typed Slim Bindings

### ISSUE THREAD

**Issue by jpobst** [jpobst/Prototype.Android.MavenBindings#4]
## Background

One of the key tenets of "slim bindings" is that you only bind the type(s)/method(s) that you need to use.  Unfortunately this does not work for a normal binding project because your desired types likely derive from other unbound types or take unbound types as method parameters or return types.

Example:
```java
public class MapView extends MapViewBase implements MappableComponent {
  public MapView (Activity activity, MapViewOptions options) { ... }

  public PinResult AddPin (string name, PinCoordinates coordinates) { ... }
}
```

Trying to bind just `MapView` would fail because it also needs the types: `MapViewBase`, `MappableComponent`, `MapViewOptions`, `PinResult`, and `PinCoordinates`.  Binding each of those types may require further types, and you essentially end up needing to bind the entire library.

## Slim Bindings

The way slim bindings fixes this is that you write a Java wrapper around the API you need that does no expose these custom types to C#.  This may look something like:

```java
public class MyMapView extends android.view.View {
  private MapView mv;

  public MyMapView (Activity activity, string option1, bool option2) {
    var options = new MapViewOptions (option1, option2);
    mv = new MapView (activity, options);
  }
  
  public void AddPin (string name, long latitude, long longitude) {
    mv.AddPin (name, new PinCoordinates (latitude, longitude);
  }
}
```

In effect, it performs "type erasure" of custom types in order to bind the class.

One potential downside to this is that users must write a Java wrapper in order to expose the correct API, and many users may not know Java or would prefer to remain in C#.

## Flutter's New `jingen` Bindings

Either as a conscience choice or because it is very early in development, `jnigen` does not expose any custom types in their bindings.  Every non-primitive type is exposed as `jni.JObject`.  (Their ~equivalent of `Java.Lang.Object`.)

So their binding of the above class would look like this (in a C#-looking language rather than dart):

```csharp
public class MyMapView : Java.Lang.Object {
  public MyMapView (Java.Lang.Object activity, Java.Lang.Object options) { ... }

  public Java.Lang.Object AddPin (string name, Java.Lang.Object coordinates) { ... }
}
```

Basically, they are performing the same "type erasure" that our slim bindings propose.
